cmd/kube-rbac-proxy/app/kube-rbac-proxy.go

```diff
@@ -263,11 +264,15 @@ func Run(cfg *server.KubeRBACProxyConfig) error {
 		<-stoppedCh
 	}()
 
+	cfg.SecureServing.Cert.CurrentCertKeyContent()
+
 	if cfg.KubeRBACProxyInfo.ProxyEndpointsSecureServing != nil {
 		// we need a second listener in order to serve proxy-specific endpoints
 		// on a different port (--proxy-endpoints-port)
 		proxyEndpointsMux := http.NewServeMux()
-		proxyEndpointsMux.HandleFunc("/healthz", func(w http.ResponseWriter, r *http.Request) { _, _ = w.Write([]byte("ok")) })
+		proxyEndpointsMux.HandleFunc("/healthz", func(w http.ResponseWriter, r *http.Request) {
```

Comment:
I think the previous `/healthz` was completely fine. When the proxy go routine would finish, it would also cancel the health endpoint.

What you added might be worth to be used as `liveness` or whatever you call it. But the question is, if this makes sense or if this is just testing go std lib. Maybe we could ask @enj, if this makes sense.

Reply:
I am unsure of exactly what this trying to do, but I do think that you need something better than just "always return ok."  For example, I would at a minimum expect a check that asserts that the upstream is working.  I don't know if checking this specific listener is the correct approach though.  From the YAML below I think this is acting as a readiness check and not a health check?